benchmarks/src/has_single_bit.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <benchmark/benchmark.h>
+#include <bit>
+#include <cstdint>
+
+#include <utility.hpp>
+
+using namespace std;
+
+template <typename T>
+void has_single_bit_if(benchmark::State& state) {
+    const auto random_v = random_vector<T>(8);
+    for (auto _ : state) {
+        benchmark::DoNotOptimize(random_v);
+        unsigned int count_true  = 0;
+        unsigned int count_false = 0;
+        for (const auto& x : random_v) {
+            if (has_single_bit(x)) {
+                benchmark::DoNotOptimize(++count_true);
+            } else {
+                benchmark::DoNotOptimize(++count_false);
+            }
+        }
+    }
+}
+
+template <typename T>
+void has_single_bit(benchmark::State& state) {

```

Comment:
This shadows the library function, which we conventionally avoid. It happened to work with google-benchmark 1.9.1, but conflicts with google-benchmark 1.9.2 in #5370.